esx_service/vmdk_ops.py

```diff
@@ -606,17 +617,23 @@ def setStatusDetached(vmdk_path):
 
 
 def getStatusAttached(vmdk_path):
-    '''Returns (attached, uuid) tuple. For 'detached' status uuid is None'''
+    '''Returns (attached, uuid, attach_as) tuple. For 'detached' status uuid is None'''
 
     vol_meta = kv.getAll(vmdk_path)
+    try:
+        attach_as = vol_meta[kv.VOL_OPTS][kv.ATTACH_AS]
+    except:
+        attach_as = kv.DEFAULT_ATTACH_AS
+
     if not vol_meta or kv.STATUS not in vol_meta:
-        return False, None
+        return False, None, attach_as
```

Comment:
Since this is getting dynamic state should this just return None for the attach_as as well.